### ISSUE TITLE
CosmosSerializer not honored in LINQ queries

### ISSUE THREAD

**Issue by MITIN33** [Azure/azure-cosmos-dotnet-v3#1936]
It seems like LINQ queries use CosmosSerializationOptions for serialization of fields, but it does not supports ENUM to String conversion. And If i use CosmosSerilaizer then it is  not honored by LINQ queries, is there a workaround **without** decorating field with [JsonConverter(typeof(StringEnumconverter)] and make LINQ queries work for ENUM fields ?

**Comment by ealsur** [comment]
@khdang Do you have any comment on this one?

**Comment by j82w** [comment]
This is a duplicate of https://github.com/Azure/azure-cosmos-dotnet-v3/issues/1782

**Comment by MITIN33** [comment]
@j82w  as mentioned in #1782 -  when i try to set both Serilazer and SerializerOption it throws ArgumentException -"'SerializerOptions is not compatible with Serializer. Only one can be set. "'

**Comment by j82w** [comment]
Please try the [3.17.0 nuget](https://www.nuget.org/packages/Microsoft.Azure.Cosmos/3.17.0).

**Comment by j82w** [comment]
Sorry for closing this. I got the issues number mixed up. This could be address with a new linq serialization option.

**Comment by Maya-Painter** [comment]
Hi all, we've merged what we hope will be a solution to this thread by allowing the custom serializer to be used for LINQ translations. This also unblocks having System.Text.Json attributes applied in LINQ queries.

We plan to release this as part of the next preview package, but I also wanted to take this time to confirm these changes meet your needs before finalizing them as part of our public API.  If you try the changes out, please follow up in the thread if this update addresses your concerns. If it doesn't, please let us know why.

To use your own custom serializer for LINQ translations you must implement `CosmosLinqSerializer` and set it as a custom serializer on the `CosmosClient`. There is a sample of this for serialization with System.Text.Json in CosmosLinqSerializer.cs.
